webapi/src/main/scala/org/knora/webapi/messages/v2/responder/valuemessages/ValueMessagesV2.scala

```diff
@@ -2820,11 +2820,65 @@ object HierarchicalListValueContentV2 extends ValueContentReaderV2[HierarchicalL
       throw BadRequestException(s"List node IRI <$listValueAsListNode> is not a Knora data IRI")
     }
 
-    HierarchicalListValueContentV2(
-      ontologySchema = ApiV2Complex,
-      valueHasListNode = listValueAsListNode.toString,
-      comment = getComment(jsonLDObject)
-    )
+    for {
+      isRootNode <- checkIfNodeIsRoot(
+                      jsonLDObject = jsonLDObject,
+                      appActor = appActor,
+                      featureFactoryConfig = featureFactoryConfig
+                    )
+
+      _ = if (isRootNode) {
+            throw BadRequestException("Root nodes cannot be set as values.")
+          }
+
+      listValueContent <-
+        Future(
+          HierarchicalListValueContentV2(
+            ontologySchema = ApiV2Complex,
+            valueHasListNode = listValueAsListNode.toString,
+            comment = getComment(jsonLDObject)
+          )
+        )
+    } yield listValueContent
+  }
+
+  private def checkIfNodeIsRoot(
```

Comment:
docstring is missing

Reply:
In this package only objects have docstrings, but in this case name speaks by itself and all arguments are inherited from the main definition.